### DISCUSSION TITLE
Getting SoC via MQTT instead of VE.Direct

### DISCUSSION THREAD

**Discussion by bblo-seso** [hoylabs/OpenDTU-OnBattery#464] [General]
Hi All,

My setup is very similar to this one from your Wiki: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/HM%E2%80%901500-%7C--4x-385W-%7C-150-35-MPPT-%7C--2x-25,6V-(51,2V)-LiFePo

My Victron Shunt & Victron Solarcharger are connected via VE.Direct to a Raspberry Pi with Venus OS. The Victron integration then pulls all Shunt & Solarcharger data nicely into Home Assistant.

This means the VE.Direct ports aren't free anymore on my Victron devices.

Question: Is there a way within OpenDTU-onBattery to receive the Battery SoC via MQTT instead of directly via the VE.Direct port?

thanks for your answers!

**Comment by stanpete78** [comment]
I'm with you - we should have more MQTT in general. I basically filed the same thing as a feature request but for PV power values. It's such a shame I'm not a programmer because I would love to get this working.

**Comment by schlimmchen** [comment]
See #293.
